connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -185,8 +185,16 @@ private void doRun() throws InterruptedException {
 
             execute();
         } catch (Throwable t) {
-            log.error("{} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted", this, t);
-            throw t;
+            if (!stopping && !cancelled) {
```

Comment:
Should we just write this as: 
```
if (cancelled) {
} else if(stopping) {
} else {
}
```
?
Should be equivalent but a bit easier to follow.

Reply:
Very good suggestion and this made the flow more simple.  I have implemented it. @kkonstantine  please check the PR again.